Cabal/doc/developing-packages.rst

```diff
@@ -0,0 +1,2424 @@
+    ``clib``:
+
+    ::
+
+        include-dirs:     cbits
+        c-sources:        clib.c
+        install-includes: clib.h
+
+    Now any package that depends (directly or transitively) on the
+    ``bindings-clib`` library can use ``clib.h``.
+
+    Note that in order for files listed in ``install-includes`` to be
+    usable when compiling the package itself, they need to be listed in
+    the ``includes:`` field as well.
+
+``include-dirs:`` *directory list*
+    A list of directories to search for header files, when preprocessing
+    with ``c2hs``, ``hsc2hs``, ``cpphs`` or the C preprocessor, and also
+    when compiling via C. Directories can be absolute paths (e.g., for
+    system directories) or paths that are relative to the top of the
+    source tree. Cabal looks in these directories when attempting to
+    locate files listed in ``includes`` and ``install-includes``.
+``c-sources:`` *filename list*
+    A list of C source files to be compiled and linked with the Haskell
+    files.
+``js-sources:`` *filename list*
+    A list of JavaScript source files to be linked with the Haskell
+    files (only for JavaScript targets).
+``extra-libraries:`` *token list*
+    A list of extra libraries to link with.
+``extra-ghci-libraries:`` *token list*
+    A list of extra libraries to be used instead of 'extra-libraries'
+    when the package is loaded with GHCi.
+``extra-lib-dirs:`` *directory list*
+    A list of directories to search for libraries.
+``cc-options:`` *token list*
+    Command-line arguments to be passed to the C compiler. Since the
+    arguments are compiler-dependent, this field is more useful with the
+    setup described in the section on `system-dependent
+    parameters <#system-dependent-parameters>`__.
+``cpp-options:`` *token list*
+    Command-line arguments for pre-processing Haskell code. Applies to
+    haskell source and other pre-processed Haskell source like .hsc
+    .chs. Does not apply to C code, that's what cc-options is for.
+``ld-options:`` *token list*
+    Command-line arguments to be passed to the linker. Since the
+    arguments are compiler-dependent, this field is more useful with the
+    setup described in the section on `system-dependent
+    parameters <#system-dependent-parameters>`__>.
+``pkgconfig-depends:`` *package list*
+    A list of
+    `pkg-config <http://www.freedesktop.org/wiki/Software/pkg-config/>`__
+    packages, needed to build this package. They can be annotated with
+    versions, e.g. ``gtk+-2.0 >= 2.10, cairo >= 1.0``. If no version
+    constraint is specified, any version is assumed to be acceptable.
+    Cabal uses ``pkg-config`` to find if the packages are available on
+    the system and to find the extra compilation and linker options
+    needed to use the packages.
+
+    If you need to bind to a C library that supports ``pkg-config`` (use
+    ``pkg-config --list-all`` to find out if it is supported) then it is
+    much prefer
```

Comment:
What do you mean? I can see it here: http://cabal.readthedocs.io/en/wip-sphinx-doc/developing-packages.html#build-information